### ISSUE TITLE
Validate required array field

### ISSUE THREAD

**Issue by torstenrudolf** [json-schema-form/angular-schema-form#45]
This is possibly related to issue #27 .

If you have an array of strings, the `required` flag is not validated.

The following will not be $invalid when the array is empty.

``` json
{
    "type": "object",
    "required": ["editors_used"],
    "properties": {
        "editors_used": {
            "type": "array",
            "title": "What editors do you use?",
            "minItems": 1,
            "items": {
                "type": "string",
                "enum": [
                    "vim", "emacs", "sublime", "kate"
                ]
            },
            "additionalItems": false
        }
    }
}
```

**Comment by ob-ivan** [comment]
I have an opposite problem: I have an array which is not required, but removing all items from it (i.e. making it an empty array) results in form being invalid.

I can open a separate issue and post a gist if needed.

**Comment by jimmykane** [comment]
Is this resolved ? I am having this issue with latest stable.

**Comment by Anthropic** [comment]
@jimmykane it looks like it should be, have you tried with the alpha?

**Comment by jimmykane** [comment]
@Anthropic Is alpha stable enough? We have our own decorators unfortunately and somehow I caught right now in the middle. 
Might the bootstrap decorators (repo link?) work or should I test also against alpha?

**Comment by Anthropic** [comment]
angular-schema-form-bootstrap repo is in this org, if you use the decorator file from there it is more recent than the bundled one in bower. Are you setting validate on render? If not it would only validate on submit.

**Comment by jimmykane** [comment]
@Anthropic Will check this out. Just saw some sf-new-array and the old one got deprecated. So let me check that.

**Comment by jimmykane** [comment]
AS far as I checked the following: 
```
{
    "type": "object",
    "required": ["editors_used"],
    "properties": {
        "editors_used": {
            "type": "array",
            "title": "What editors do you use?",
            "minItems": 1,
            "items": {
                "type": "string",
                "enum": [
                    "vim", "emacs", "sublime", "kate"
                ]
            },
            "additionalItems": false
        }
    }
}
```
Will validatate for the whole form but not for the checkboxes. The form state is invalid but only after a change is triggered the checkboxes get the correct validity state

**Comment by jimmykane** [comment]
Here is the plnkr as well https://plnkr.co/edit/fjSB0WgkJ46GuQZW6sz8?p=preview

The problem is still there but not on alpha

Tested -> Alpha is ok
Tested -> Old .8 with new decorators not OK.
